### PR TITLE
OpenInferenceCallbackHandler uses node IDs rather than hashes

### PR DESCRIPTION
**Pull request by axiomofjoy**
# Description

Change the `OpenInferenceCallbackHandler` to use node IDs rather than node hashes. Node hashes were originally chosen because of an issue reading in legacy LlamaIndex persisted indexes, but the node IDs work with newer versions.

This issue was initially investigated [here](https://github.com/jerryjliu/llama_index/issues/7101), where we found that the issue disppear after re-indexing with a newer version of LlamaIndex.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by axiomofjoy** [comment]
@logan-markewich Please wait to merge this one in until I am able to make corresponding changes to the tutorial notebooks on our side 😄

**Comment by logan-markewich** [comment]
Hey @axiomofjoy is this safe to merge yet?

**Comment by axiomofjoy** [comment]
@logan-markewich Thanks for the reminder. Not yet, I still need to update the notebook on our side.

File: llama_index/callbacks/open_inference_callback.py [hunk 1]
```diff
@@ -208,18 +208,22 @@ def on_event_end(
...
-            self._trace_data.query_data.response_text = str(
-                payload.get(EventPayload.RESPONSE, "")
-            ) or str(payload.get(EventPayload.COMPLETION, ""))
+            if chat_response := payload.get(EventPayload.RESPONSE):
+                response_text = chat_response.message.content
+            elif completion_response := payload.get(EventPayload.COMPLETION):
+                response_text = completion_response.text
+            else:
+                response_text = ""
+            self._trace_data.query_data.response_text = response_text
```
Comment (axiomofjoy) [inline llama_index/callbacks/open_inference_callback.py:226]: The role is being included as part of the response text when calling `str` on an instance of `ChatResponse` or `CompletionResponse` directly. So all of the response text is of the form "assistant: ..."

**Comment by pRivAte12** [comment]
Hi, @axiomofjoy 

I had an issue with using Arize/Phoenix since `OpenInferenceCallback` returns the Node hash as an ID.
It seems similar to this issue: https://github.com/run-llama/llama_index/issues/7101

I assume `NodeData` should be created with the Node ID instead of the hash, same with the changes made in this PR. Are there any plans to reopen this PR?

I temporarily forked Llama-index and made changes to OpenInferenceCallback when using it with [Phoenix](https://github.com/Arize-ai/phoenix). I hope this can be fixed in the future.

**Pull request closed without merging**